src/main/java/org/folio/rest/service/holdings/HoldingsServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+package org.folio.rest.service.holdings;
+
+import static org.folio.rest.repository.holdings.LoadStatus.COMPLETED;
+import static org.folio.rest.repository.holdings.LoadStatus.IN_PROGRESS;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import org.folio.holdingsiq.model.Holding;
+import org.folio.holdingsiq.model.HoldingsLoadStatus;
+import org.folio.rest.repository.holdings.HoldingsRepository;
+import org.folio.rest.repository.holdings.LoadStatus;
+import org.folio.rest.util.template.RMAPITemplateContext;
+
+@Component
+public class HoldingsServiceImpl implements HoldingsService {
+
+  private static final int MAX_COUNT = 5000;
+  private static final Logger logger = LoggerFactory.getLogger(HoldingsServiceImpl.class);
+  private Vertx vertx;
+  private long delay;
+  private HoldingsRepository holdingsRepository;
+
+  @Autowired
+  public HoldingsServiceImpl(Vertx vertx, HoldingsRepository holdingsRepository, @Value("${holdings.status.check.delay}") long delay) {
+    this.vertx = vertx;
+    this.holdingsRepository = holdingsRepository;
+    this.delay = delay;
+  }
+
+  public CompletableFuture<Void> loadHoldings(RMAPITemplateContext context, String tenantId) {
+    return populateHoldings(context)
+      .thenCompose(isSuccessful -> waitForCompleteStatus(context, 5, tenantId))
```

Comment:
Should parameters such as (5) be store in application.properties (as you did for delay)